contracts/staking/LPStaker.sol

```diff
@@ -0,0 +1,157 @@
+        require(msg.sender == router || msg.sender == user, "Access denied");
+        _;
+    }
+
+    modifier onlyRouter() {
+        require(msg.sender == router, "Access denied");
+        _;
+    }
+
+    constructor(address _router, address _stakingRewardsFactory) public {
+        router = _router;
+        stakingRewardsFactory = _stakingRewardsFactory;
+    }
+
+    function balanceOf(address user, address pool) external view returns (uint256) {
+        return userVsPoolVsbalances[user][pool];
+    }
+
+    function stake(address user, address pool, uint256 amount) external onlyRouter {
+        updateReward(user, pool);
+        (address stakingRewardsAddress, , ) = IStakingRewardsFactory(stakingRewardsFactory).stakingRewardsInfoByStakingToken(pool);
+
+        require(stakingRewardsAddress != address(0), "No staking rewards exists for the given pool");
+
+        IStakingRewards stakingRewards = IStakingRewards(stakingRewardsAddress);
+
+        userVsPoolVsbalances[user][pool] = userVsPoolVsbalances[user][pool].add(amount);
+
+        TransferHelper.safeApprove(pool, stakingRewardsAddress, amount);
+        stakingRewards.stake(amount);
+
+        emit LPStaked(user, pool, amount);
+    }
+
+    function withdraw(address user, address pool, uint256 amount) external onlyRouterOrUser(user) {
+        updateReward(user, pool);
+
+        (address stakingRewardsAddress, , ) = IStakingRewardsFactory(stakingRewardsFactory).stakingRewardsInfoByStakingToken(pool);
+
+        require(stakingRewardsAddress != address(0), "No staking rewards exists for the given pool");
+        
+        IStakingRewards stakingRewards = IStakingRewards(stakingRewardsAddress);
+
+        userVsPoolVsbalances[user][pool] = userVsPoolVsbalances[user][pool].sub(amount);
+
+        stakingRewards.withdraw(amount);
+
+        TransferHelper.safeTransfer(pool, msg.sender, amount);
+
+        emit LPWithdrawn(
+            user,
+            pool,
+            amount
+        );
+    }
+
+    function getReward(address user, address pool) external onlyRouterOrUser(user) returns(address, uint256){
+
+        updateReward(user, pool);
+
+        (address stakingRewardsAddress, , ) = IStakingRewardsFactory(stakingRewardsFactory).stakingRewardsInfoByStakingToken(pool);
+
+        require(stakingRewardsAddress != address(0), "No staking rewards exists for the given pool");
+        
+        IStakingRewards stakingRewards = IStakingRewards(stakingRewardsAddress);
+
+        uint256 reward = userVsPoolVsRewards[user][pool];
+
+        address rewardToken = stakingRewards.rewardsToken();
+
+        if (reward > 0) {
+            userVsPoolVsRewards[user][pool] = 0;
+
+            stakingRewards.getReward();
+            
+            TransferHelper.safeTransfer(rewardToken, msg.sender, reward);
+
+            emit LPRewardPaid(user, pool, reward);
+
+            return (rewardToken, reward);
+        }
+
+        return (rewardToken, reward);
+    }
+
```

Comment:
There is no `exit()` function for withdrawing and claiming in the same transaction.

Reply:
We don't need an exit function here. The LPStaker contract is used by the Router contract. Router contract calls withdraw and getRewards methods one after another.
@mudgen